examples/pytorch/diffpool/train.py

```diff
@@ -0,0 +1,323 @@
+    model = DiffPool(input_dim, 
+                     hidden_dim, 
+                     embedding_dim,  
+                     label_dim,
+                     activation, 
+                     prog_args.gc_per_block,  
+                     prog_args.dropout, 
+                     prog_args.num_pool, 
+                     prog_args.linkpred,
+                     prog_args.batch_size, 
+                     'maxpool', 
+                     assign_dim,
+                     prog_args.pool_ratio)
+    
+    if prog_args.load_epoch >= 0 and prog_args.save_dir is not None:
+        model.load_state_dict(torch.load(prog_args.save_dir + "/" + prog_args.dataset\
+                                         + "/model.iter-" + str(prog_args.load_epoch)))
+
+    print("model init finished")
+    print("MODEL:::::::", prog_args.method)
+    if prog_args.cuda:
+        model = model.cuda()
+    
+    logger = train(train_dataloader, model, prog_args, val_dataset=val_dataloader)
+    result = evaluate(test_dataloader, model, prog_args, logger)
+    print("test  accuracy {}%".format(result*100))
+
+def collate_fn(batch):
+    '''
+    collate_fn for dataset batching
+    transform ndata to tensor (in gpu is available)
+    '''
+    graphs, labels = map(list, zip(*batch))
+    #cuda = torch.cuda.is_available()
+
+    # batch graphs and cast to PyTorch tensor
+    for graph in graphs:
+        for (key, value) in graph.ndata.items():
+            graph.ndata[key] = torch.FloatTensor(value)
+    batched_graphs = dgl.batch(graphs)
+
+    # cast to PyTorch tensor
+    batched_labels = torch.LongTensor(np.array(labels))
+
+    return batched_graphs, batched_labels
+
+def train(dataset, model, prog_args, same_feat=True, val_dataset=None):
+    '''
+    training function
+    '''
+    dir = prog_args.save_dir + "/" + prog_args.dataset
+    if not os.path.exists(dir):
+        os.makedirs(dir)
+    dataloader = dataset
+    optimizer = torch.optim.Adam(filter(lambda p: p.requires_grad,
+                                        model.parameters()), lr=0.001)
+    early_stopping_logger = {"best_epoch":-1, "val_acc": -1}
+
+    if prog_args.cuda > 0:
+        torch.cuda.set_device(0)
+    for epoch in range(prog_args.epoch):
+        begin_time = time.time()
+        model.train()
+        train_accu = 0
+        print("EPOCH ###### {} ######".format(epoch))
+        computation_time = 0.0
+        for (batch_idx, (batch_graph, graph_labels)) in enumerate(dataloader):
+            if torch.cuda.is_available():
+                for (key, value) in batch_graph.ndata.items():
+                    batch_graph.ndata[key] = value.cuda()
+                graph_labels = graph_labels.cuda()
+
+                
+            model.zero_grad()
+            compute_start = time.time()
+            ypred = model(batch_graph)
+            indi = torch.argmax(ypred, dim=1)
+            correct = torch.sum(indi == graph_labels).item()
+            train_accu += correct
```

Comment:
change name to train_true_sample or something else. 
Add `train_count += correct.shape[0]`